### PR TITLE
Adding nvidia as runtime for containers requiring GPU

### PR DESCRIPTION
**Pull request by ubhattacharjya**
### Summary

This change adds runtime as "nvidia" to docker host config of containers requiring GPU.

### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/api/task/task.go [hunk 1]
```diff
@@ -309,6 +310,14 @@ func (task *Task) populateGPUEnvironmentVariables() {
 	}
 }
 
+func (task *Task) shouldRequireNvidiaRuntime(container *apicontainer.Container) bool {
+	_, present := container.Environment[nvidiaVisibleDevicesEnvVar]
```
Comment (sharanyad) [inline agent/api/task/task.go]: `ok` should be good enough :)

File: agent/api/task/task.go [hunk 2]
```diff
@@ -921,6 +930,11 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 		Resources:    resources,
 	}
 
+	requireNvidiaRuntime := task.shouldRequireNvidiaRuntime(container)
+	if requireNvidiaRuntime {
```
Comment (sharanyad) [inline agent/api/task/task.go]: this can just be:
`if task.shouldRequireNvidiaRuntime(container) {`

File: agent/api/task/task.go [hunk 3]
```diff
@@ -921,6 +930,11 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
...
+		hostConfig.Runtime = nvidiaRuntime
```
Comment (sharanyad) [inline agent/api/task/task.go]: Can you also add a log line indicating that the runtime is set as nvidia?

File: agent/api/task/task.go [hunk 4]
```diff
@@ -309,6 +310,14 @@ func (task *Task) populateGPUEnvironmentVariables() {
 	}
 }
 
+func (task *Task) shouldRequireNvidiaRuntime(container *apicontainer.Container) bool {
+	_, ok := container.Environment[nvidiaVisibleDevicesEnvVar]
+	if ok {
```
Comment (petderek) [inline agent/api/task/task.go]: `return ok`

**Pull request merged**